var/spack/repos/builtin/packages/microsocks/package.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Microsocks(MakefilePackage):
+    """Microsocks is a multithreaded, small, efficient SOCKS5 server.
+    It is a SOCKS5 service that you can run on your remote boxes to
+    tunnel connections through them, if for some reason SSH doesn't
+    cut it for you."""
+
+    homepage = "https://github.com/rofl0r/microsocks"
+    url      = "https://github.com/rofl0r/microsocks/archive/refs/tags/v1.0.2.tar.gz"
+    git      = "https://github.com/rofl0r/microsocks.git"
+
+    maintainers = ["jcpunk"]
+
+    version("develop", branch="master")
+    version("1.0.2", sha256="5ece77c283e71f73b9530da46302fdb4f72a0ae139aa734c07fe532407a6211a")
+
+    def _make(self, *args, **kwargs):
+        make(
+            "VERBOSE=1",
+            "prefix=" + self.prefix,
+            "CFLAGS=" + self.compiler.c99_flag,
+            *args,
+            **kwargs
+        )
+
+    def build(self, spec, prefix):
+        self._make("microsocks")
```

Comment:
If you set `build_targets = ['CFLAGS={0}'.format(self.c99_flag), 'microsocks'] and `install_targets = ['CFLAGS={0}'.format(self.c99_flag), 'install'], you may not have to define `_make`, `build`, or `install`. You can also set CFLAGS like in `hpcc` (that package defines a `flag_handler` method, which is handled specially by Spack).

That isn't essential but would make this package a bit more idiomatic WRT other Spack make-based packages - would you be able to update it?

Reply:
Is there an example I can look at?  I mostly copied this one from `zstd`.

Reply:
The `hpcc` package has an example of defining a `flag_handler` method, in which case you would not have to set `build_targets` or `install_targets` (and you would also not have to define `_make`, `build`, or `install`. Let me know if that doesn't cover it.

Reply:
I'm not seeing a clear way to set `prefix=` (note lower case) as an element of the `install` step with the `flag_handler` method....

Reply:
https://spack.readthedocs.io/en/latest/packaging_guide.html?highlight=flag_handler#compiler-flags will explain which of those elements to leave as `None` and which one you'll want to set. For example if you want to export CFLAGS as an environment variable, you would return `(None, flags, None)`.

> I'm not seeing a clear way to set prefix= (note lower case) as an element of the install step with the flag_handler method....

Sorry I might have missed something:

* As you point out, you can't set prefix in `flag_handler`
* Spack already defines an `install` method for `MakefilePackage`s and it already tries to set `PREFIX`, but maybe the issue is that this makefile only understands `prefix` (lower-case) so in that case, you would have to define `install` as you did

Reply:
Ok, I think I've got it.  I used the trick from `unzip/package.py` to get the `install_targets` to have the right prefix set.

Does this latest patch look right?

Reply:
Yes - thanks!